### ISSUE TITLE
enabling AAPT2 during build is causing some problems to some Android versions/flavors

### ISSUE THREAD

**Issue by arielenter** [siacs/Conversations#2785]
Are you aware of the following issue that is affecting some users?

https://gitlab.com/fdroid/fdroiddata/issues/979

Today I tried Conversations 1.23.4 from f-droid and it seems that problem is present on it too.

For what I understand, It seems that f-droid won't be taking action and that it is suggested that we should contact the affected apps' developers for help on the matter.

Thank you.

**Comment by hre999** [comment]
can confirm, its crashing since de23b37

Its a common issue with getPkgName on OEM and custom ROMs:
https://issuetracker.google.com/issues/64434571

In short: Some ROMS don't copy a buffer in getPkgName, which with aapt is fine, with aapt2 will produce crashes.

Actually a bug with the ROMs, but its easily fixed with adding `android.enableAapt2=false` to _grade.properties_ or using build tools >=27 (the latter I did not test though)

**Comment by iNPUTmice** [comment]
This APK is build with 27.0.3: https://gultsch.de/files/Conversations-1.23.5-playstore-release.apk - please have a look if it fixes your issues. Otherwise I'll have to downgrade aapt

**Comment by STPKITT** [comment]
Thanks for the fast action iNPUTmice!
I tried the 1.23.5 APK on a Samsung Galaxy Grand DUOS GT-I9082 running LineageOS 13.0 6.0.1_r78 from July 2017 and to me the problem seems to be solved. The only issue I experienced was Conversations crashing when leaving the "About Conversations"-screen via the hardware back button. The crash does not happen when leaving that screen via the onscreen back icon in the top left of the screen. I've send in the crash report via Conversations' onboard bug reporting system.
